packages/react-examples/src/react/FocusTrapZone/e2e/FocusTrapZone.e2e.ts

```diff
@@ -1,24 +1,413 @@
+      setProps({ disableFirstFocus: true });
+
+      // For some reason this doesn't work:
+      // cy.focused().should('match', 'body');
+      cy.document().should(doc => {
+        expect(doc.activeElement?.tagName).to.equal('BODY');
+      });
+    });
+
+    it('Does not focus first on mount while disabled', () => {
+      setProps({ disabled: true });
+
+      cy.document().should(doc => {
+        expect(doc.activeElement?.tagName).to.equal('BODY');
+      });
+    });
+
+    it('Focuses on firstFocusableSelector on mount', () => {
+      // deprecated: this is actually a className, not a selector
+      setProps({ firstFocusableSelector: 'last-class' });
+
+      cy.focused().should('have.id', 'last');
+    });
+
+    it('Does not focus on firstFocusableSelector on mount while disabled', () => {
+      setProps({ firstFocusableSelector: 'last-class', disabled: true });
+
+      cy.document().should(doc => {
+        expect(doc.activeElement?.tagName).to.equal('BODY');
+      });
+    });
+
+    it('Falls back to first focusable element with invalid firstFocusableSelector', () => {
+      setProps({ firstFocusableSelector: 'invalidSelector' });
+
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('Focuses on firstFocusableTarget selector on mount', () => {
+      setProps({ firstFocusableTarget: '#last' });
+
+      cy.focused().should('have.id', 'last');
+    });
+
+    it('Focuses on firstFocusableTarget callback on mount', () => {
+      setProps({ firstFocusableTarget: element => element.querySelector('#last') });
+
+      cy.focused().should('have.id', 'last');
+    });
+
+    it('Does not focus on firstFocusableTarget selector on mount while disabled', () => {
+      setProps({ firstFocusableTarget: '#last', disabled: true });
+
+      cy.document().should(doc => {
+        expect(doc.activeElement?.tagName).to.equal('BODY');
+      });
+    });
+
+    it('Does not focus on firstFocusableTarget callback on mount while disabled', () => {
+      setProps({
+        firstFocusableTarget: (element: HTMLElement) => element.querySelector('#last'),
+        disabled: true,
+      });
+
+      cy.document().should(doc => {
+        expect(doc.activeElement?.tagName).to.equal('BODY');
+      });
+    });
+
+    it('Falls back to first focusable element with invalid firstFocusableTarget selector', () => {
+      setProps({ firstFocusableTarget: '.invalidSelector' });
+
+      cy.focused().should('have.id', 'first');
+    });
+
+    it('Falls back to first focusable element with invalid firstFocusableTarget callback', () => {
+      setProps({ firstFocusableTarget: () => null });
+
+      cy.focused().should('have.id', 'first');
+    });
+  });
+
   describe('Tab and shift-tab wrap at extreme ends of the FTZ', () => {
+    // Note: all of the IDs in these tests refer to buttons
+
+    it('can tab across a FocusZone with different button structures', () => {
```

Comment:
This test name is completely alien to me. Are you testing that it's possible to tab in/out of a FZ inside a FTZ ?

Reply:
Also what's a 'different button structure' ?

Reply:
I mostly copied the descriptions from the previous tests... 😬  I *think* what you said is basically what it's testing.